turbinia/api/routes/evidence.py

```diff
@@ -0,0 +1,259 @@
+  raise HTTPException(status_code=404, detail='No evidence found.')
+
+
+@router.get('/query')
+async def query_evidence(
+    request: Request, attribute_name: str = Query(
+        'request_id', enum=EVIDENCE_QUERY_ATTRIBUTES),
+    attribute_value: str = Query(), output: str = Query(
+        'keys', enum=('keys', 'values', 'count'))):
+  """Queries evidence in Redis that have the specified attribute value.
+
+  Args:
+    attribute_name (str): Name of attribute to be queried.
+    attribute_value (str): Value the attribute must have.
+    output Optional(str): Sets how the evidence found will be output.
+
+  Returns:
+    summary (dict): Summary of all evidences and their content.
+  
+  Raises:
+    HTTPException: If no matching evidence is found.
+  """
+  if attribute_name and attribute_name not in EVIDENCE_QUERY_ATTRIBUTES:
+    raise HTTPException(
+        status_code=400, detail=(
+            f'Cannot query by {attribute_name}. '
+            f'Queryable attributes: {EVIDENCE_QUERY_ATTRIBUTES}'))
+  evidences_found = None
+  if attribute_value == 'hash':
+    evidences_found = redis_manager.get_evidence_key_by_hash(attribute_name)
+  else:
+    evidences_found = redis_manager.query_evidence(
+        attribute_name, attribute_value, output)
+  if evidences_found:
+    return JSONResponse(content=evidences_found, status_code=200)
+  raise HTTPException(
+      status_code=404, detail=(
+          f'No evidence found with value {attribute_value} in attribute '
+          f'{attribute_name}.'))
+
+
+@router.get('/{evidence_id}')
+async def get_evidence_by_id(request: Request, evidence_id):
+  """Retrieves an evidence in Redis by using its UUID.
+
+  Args:
+    evidence_id (str): The UUID of the evidence.
+  
+  Raises:
+    HTTPException: if the evidence is not found.
+
+  Returns:
+    Dictionary of the stored evidence
+  """
+  if stored_evidence := redis_manager.get_evidence_data(evidence_id):
+    return JSONResponse(content=stored_evidence, status_code=200)
+  raise HTTPException(
+      status_code=404,
+      detail=f'UUID {evidence_id} not found or it had no associated evidences.')
+
+
+@router.post('/upload')
+async def upload_evidence(
+    ticket_id: Annotated[str, Form()], files: List[UploadFile],
+    calculate_hash: Annotated[bool, Form()] = False):
+  """Upload evidence file to server for processing.
+
+  Args:
+    ticket_id (str): ID of the ticket, which will be the name of the folder 
+      where the evidence will be saved.
+    calculate_hash (bool): Boolean defining if the hash of the evidence should
+      be calculated.
+    file (List[UploadFile]): Evidence files to be uploaded to folder for later
+        processing. The maximum size of the file is 10 GB. 
+  
+  Returns:
+    List of uploaded evidences or warning messages if any.
+  """
+  evidences = []
+  for file in files:
+    warning_message = None
+    try:
+      file_path = await get_file_path(file.filename, ticket_id)
```

Comment:
file_path would not be accessible below if an OSError exception is thrown. I would either initialize file_path to None or consolidate both try statements in one.